Framework/src/PostProcessingDevice.cxx

```diff
@@ -0,0 +1,133 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   PostProcessingDevice.cxx
+/// \author Piotr Konopka
+///
+
+#include "QualityControl/PostProcessingDevice.h"
+
+#include "QualityControl/PostProcessingRunner.h"
+#include "QualityControl/QcInfoLogger.h"
+
+#include <Common/Exceptions.h>
+#include <Configuration/ConfigurationFactory.h>
+#include <Framework/CallbackService.h>
+#include <Framework/ControlService.h>
+
+using namespace AliceO2::Common;
+using namespace o2::configuration;
+using namespace o2::framework;
+
+constexpr auto outputBinding = "mo";
+
+namespace o2::quality_control::postprocessing
+{
+
+PostProcessingDevice::PostProcessingDevice(const std::string& taskName, const std::string& configurationSource)
+  : mRunner(std::make_unique<PostProcessingRunner>(taskName)), mDeviceName(createPostProcessingIdString() + "-" + taskName), mConfigSource(configurationSource)
+{
+  ILOG_INST.setFacility("PostProcessing");
+}
+
+void PostProcessingDevice::init(framework::InitContext& ctx)
+{
+  // todo: eventually we should retrieve the configuration from context
+  auto config = ConfigurationFactory::getConfiguration(mConfigSource);
+  mRunner->init(config->getRecursive());
+
+  // registering state machine callbacks
+  ctx.services().get<CallbackService>().set(CallbackService::Id::Start, [this]() { start(); });
+  ctx.services().get<CallbackService>().set(CallbackService::Id::Stop, [this]() { stop(); });
+  ctx.services().get<CallbackService>().set(CallbackService::Id::Reset, [this]() { reset(); });
+}
+
+void PostProcessingDevice::run(framework::ProcessingContext& ctx)
+{
+  // todo: enable once check runners can store the results.
```

Comment:
is this todo relevant to this class ?

Reply:
weird, i must have messed something up with the commits, since it is not enabled

Reply:
ah no, it is fine. This part is done in the next PR, at this stage we still store objects directly to the repository inside PostProcessing, because we don't send them to checkers yet. This part is correct.